app/main/page.tsx

```diff
@@ -63,15 +58,27 @@ function App() {
     foregroundDrawer.drawHandWave(badge);
   };
 
-  useEffect(() => {
-    fetchNametagFromDB()
-      .then((newNameTag) => {
-        if (newNameTag !== undefined) {
-          setNameTagContent(newNameTag);
+  const fetchUser = async () => {
+    fetchUserFromDB()
+      .then((user) => {
+        if (user.nameTag !== undefined) {
+          setNameTagContent(user.nameTag);
         }
-        setNameTagIsLoaded(true);
+        if (user.waveHands !== undefined) {
+          setWaveHandButtons(user.waveHands);
+        }
+        setHasError(false);
+        setIsLoading(false);
       })
-      .catch((err) => console.error(err));
+      .catch((err) => {
+        console.error(err);
+        setHasError(true);
+        setIsLoading(false);
+      });
```

Comment:
The advantage of using an `async` function is that you can use normal control flow (like `try-catch` blocks, etc.) with await at the places where you need to unwrap a `Promise`.